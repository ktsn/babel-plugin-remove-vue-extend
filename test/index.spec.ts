import babel = require('babel-core')
import plugin from '../src/index'

describe('Remove vue extend', () => {
  it('should remove Vue.extend', () => {
    const input = `
    import Vue from 'vue'

    export default Vue.extend({
      data () {
        return { message: 'Hello!' }
      }
    })
    `
    const { code } = babel.transform(input, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  it('should not remove local binding', () => {
    const input = `
    const Vue = {
      extend () {}
    }

    export default Vue.extend({
      data () {
        return { message: 'Hello!' }
      }
    })
    `
    const { code } = babel.transform(input, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  it('should not remove a non Ctor method', () => {
    const input = `
    import Vue from 'vue'

    export default Vue.something.extend({
      data () {
        return { message: 'Hello!' }
      }
    })
    `
    const { code } = babel.transform(input, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  it('should not remove a function', () => {
    const input = `
    import extend from './extend'

    export default extend({
      data () {
        return { message: 'Hello!' }
      }
    })
    `
    const { code } = babel.transform(input, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })

  it('should remove but move into extends option', () => {
    const input = `
    import Sub from './Sub.vue'

    export default Sub.extend({
      data () {
        return { message: 'Hello!' }
      }
    })
    `
    const { code } = babel.transform(input, {
      plugins: [plugin]
    })
    expect(code).toMatchSnapshot()
  })
})
