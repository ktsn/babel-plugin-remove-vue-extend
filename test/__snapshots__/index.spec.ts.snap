// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remove vue extend should not remove a function 1`] = `
"
import extend from './extend';

export default extend({
  data() {
    return { message: 'Hello!' };
  }
});"
`;

exports[`Remove vue extend should not remove a non Ctor method 1`] = `
"
import Vue from 'vue';

export default Vue.something.extend({
  data() {
    return { message: 'Hello!' };
  }
});"
`;

exports[`Remove vue extend should not remove local binding 1`] = `
"
const Vue = {
  extend() {}
};

export default {
  data() {
    return { message: 'Hello!' };
  }
};"
`;

exports[`Remove vue extend should remove Vue.extend 1`] = `
"

export default {
  data() {
    return { message: 'Hello!' };
  }
};"
`;

exports[`Remove vue extend should remove but move into extends option 1`] = `
"
import Sub from './Sub.vue';

export default {
  extends: Sub,

  data() {
    return { message: 'Hello!' };
  }
};"
`;
